File: Baraja.java
package Controller;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Baraja {   //creamos carta, constructor ,setter y toString
    class Carta {
        private String valor;
        private String palo;
        int valorNumerico;

        public Carta(String valor, String palo, int valorNumerico) {
            this.valor = valor;
            this.palo = palo;
            this.valorNumerico = valorNumerico;
        }

        @Override
        public String toString() {
            return valor + " de " + palo;
        }
    }

    private List<Carta> cartas;

    //Creamos la baraja y asignamos valores
    public Baraja() {
        cartas = new ArrayList<>();
        String[] valores = {"AS", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"};
        String[] palos = {"Corazones", "Diamantes", "Tréboles", "Picas"};

        for (String palo : palos) {
            for (int i = 0; i < valores.length; i++) {
                int valorNumerico = (i < 9) ? i + 1 : 10; // Asigna valores numéricos
                cartas.add(new Carta(valores[i], palo, valorNumerico));
            }
        }
    }

    //barajamos las cartas

    public void barajar() {
        Collections.shuffle(cartas);
    }

    public Carta repartirCarta() {
        return cartas.remove(0);
    }

    public void mostrarBaraja() {
        for (Carta carta : cartas) {
            System.out.println(carta);
        }
    }
}

File: BlackJack_C#/ConsoleApp1/Baraja.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Baraja
{
    public class Carta
    {
        public string Valor { get; set; }
        public string Palo { get; set; }
        //public int ValorNumerico { get; set; }

        public Carta(string valor, string palo)
        {
            this.Valor = valor;
            this.Palo = palo;
            //this.ValorNumerico = valorNumerico;
        }

        public override string ToString()
        {
            return $"{Valor} de {Palo}";
        }
    }

    private List<Carta> cartas;

    public Baraja()
    {
        cartas = new List<Carta>();
        string[] valores = { "AS", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K" };
        string[] palos = { "Corazones", "Diamantes", "Tréboles", "Picas" };

        foreach (string palo in palos)
        {
            for (int i = 0; i < valores.Length; i++)
            {
                //int valorNumerico = (i < 9) ? i + 1 : 10;
                cartas.Add(new Carta(valores[i], palo));
            }
        }
    }

    public void Barajar()
    {
        Random rng = new Random();
        cartas = cartas.OrderBy(a => rng.Next()).ToList();
    }

    public Carta RepartirCarta()
    {
        Carta carta = cartas[0];
        cartas.RemoveAt(0);
        return carta;
    }

    public void MostrarBaraja()
    {
        foreach (Carta carta in cartas)
        {
            Console.WriteLine(carta);
        }
    }
}